src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/CreditorsWithinOneYearController.java

```diff
@@ -0,0 +1,108 @@
+@PreviousController(OtherAccountingPolicyController.class)
+@RequestMapping("/company/{companyNumber}/transaction/{transactionId}/company-accounts/{companyAccountsId}/small-full/creditors-within-one-year")
+public class CreditorsWithinOneYearController extends BaseController implements
+    ConditionalController {
+
+  @Autowired
+  private CreditorsWithinOneYearService creditorsWithinOneYearService;
+
+  @Autowired
+  private BalanceSheetService balanceSheetService;
+
+  @Override
+  protected String getTemplateName() {
+    return "smallfull/creditorsWithinOneYear";
+  }
+
+  @GetMapping
+  public String getCreditorsWithinOneYear(@PathVariable String companyNumber,
+      @PathVariable String transactionId, @PathVariable String companyAccountsId, Model model,
+      HttpServletRequest request) {
+
+    addBackPageAttributeToModel(model, companyNumber, transactionId, companyAccountsId);
+
+    try {
+      CreditorsWithinOneYear creditorsWithinOneYear =
+          creditorsWithinOneYearService.getCreditorsWithinOneYear(transactionId, companyAccountsId,
+              companyNumber);
+
+      model.addAttribute("creditorsWithinOneYear", creditorsWithinOneYear);
+    } catch (ServiceException e) {
+      LOGGER.errorRequest(request, e.getMessage(), e);
+      return ERROR_VIEW;
+    }
+    return getTemplateName();
+  }
+
+  @PostMapping
+  public String postCreditorsWithinOneYear(
+      @PathVariable String companyNumber,
+      @PathVariable String transactionId,
+      @PathVariable String companyAccountsId,
+      @ModelAttribute("creditorsWithinOneYear") @Valid CreditorsWithinOneYear creditorsWithinOneYear,
+      BindingResult bindingResult, Model model, HttpServletRequest request) {
+
+    addBackPageAttributeToModel(model, companyNumber, transactionId, companyAccountsId);
+
+    if (bindingResult.hasErrors()) {
+      return getTemplateName();
+    }
+
+    try {
+      List<ValidationError> validationErrors =
+          creditorsWithinOneYearService.submitCreditorsWithinOneYear(transactionId,
+              companyAccountsId, creditorsWithinOneYear, companyNumber);
+
+      if (!validationErrors.isEmpty()) {
+        bindValidationErrors(bindingResult, validationErrors);
+        return getTemplateName();
+      }
+    } catch (ServiceException e) {
+      LOGGER.errorRequest(request, e.getMessage(), e);
+      return ERROR_VIEW;
+    }
+
+    return navigator.getNextControllerRedirect(this.getClass(), companyNumber, transactionId,
+        companyAccountsId);
+  }
+
+  @Override
+  public boolean willRender(String companyNumber, String transactionId, String companyAccountsId) {
+    try {
+      BalanceSheet balanceSheet =
+          balanceSheetService.getBalanceSheet(transactionId, companyAccountsId, companyNumber);
+
+      return balanceSheet.getOtherLiabilitiesOrAssets() != null
+          && balanceSheet.getOtherLiabilitiesOrAssets().getCreditorsDueWithinOneYear() != null;
+    } catch (ServiceException e) {
```

Comment:
The `willRender` method has been refactored and merged into develop, which you'll need to merge into this branch. The method now throws a `ServiceException`, so there's no need to catch it explicitly